### PR TITLE
feat: generate deepcopy methods for all the types

### LINKED ISSUE #1442: Consider generating DeepCopy method for all types
**Describe the feature request**
Currently in a kubebuilder project, the custom resource cannot contain field of type `*"istio.io/api/<path>".<Type>`, like `*"istio.io/api/networking/v1alpha3".VirtualService`, because of the missing `DeepCopy()` method required by [`controller-gen`](https://github.com/kubernetes-sigs/controller-tools/tree/master/cmd/controller-gen).

Steps To reproduce:
1. `cat api.go`
```
$ cat api.go
// +kubebuilder:object:generate=true
package v2

import (
        nwv1alpha3 "istio.io/api/networking/v1alpha3"
        metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

// +kubebuilder:object:root=true

// App is the Schema for the API
type App struct {
        metav1.TypeMeta   `json:",inline"`
        metav1.ObjectMeta `json:"metadata,omitempty"`

        Spec AppSpec `json:"spec"`
}

type AppSpec struct {
        VsSpec *nwv1alpha3.VirtualService `json:"vsSpec"`
}
```
2. run `controller-gen object paths=./...`

3. `go build`
```
$ go build
# dg/v2
./zz_generated.deepcopy.go:42:15: (*in).DeepCopy undefined (type *v1alpha3.VirtualService has no field or method DeepCopy)
```

4. in the generated file
```go
...
// DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
func (in *AppSpec) DeepCopyInto(out *AppSpec) {
	*out = *in
	if in.VsSpec != nil {
		in, out := &in.VsSpec, &out.VsSpec
		*out = (*in).DeepCopy()
	}
}
...
```

**Describe alternatives you've considered**
declare a new type for the corresponding istio type, implement the `DeepCopyInto()`  method manually and have `controller-gen` generate the `DeepCopy()` method for the new type.

[x] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by knight42**
Fixes #1442

NOTE: not sure if I need to wait for the update of the [common files](https://github.com/istio/api/blob/master/common/.commonfiles.sha), now I generate these methods using a local built build-tools image.

cc @jasonwzm

### CODE REVIEW

File: Makefile.core.mk [hunk 1]
```diff
@@ -219,6 +219,10 @@ $(operator_v1alpha1_pb_gos) $(operator_v1alpha1_pb_doc) $(operator_v1alpha1_pb_p
 	@$(protoc) $(go_plugin) $(protoc_gen_docs_plugin)$(operator_v1alpha1_path) $(protoc_gen_python_plugin) $^
 	@$(protoc) $(protoc_gen_k8s_support_plugins) $(operator_v1alpha1_path)/operator.proto
 	@cp -r /tmp/istio.io/api/operator/* operator
+	@sed -i -E '/DeepCopyInto supports using TypeMapStringInterface2/,+5d' $(operator_v1alpha1_path)/operator_deepcopy.gen.go
```
Comment (knight42) [inline Makefile.core.mk:222]: I am a bit curious about the usage of `TypeMapStringInterface2` and `TypeInterface2`.

it really takes me some time to figure out why the auto generated methods appear in `operator_deepcopy.gen.go` but not `operator_json.gen.go`.

**Comment by knight42** [comment]
@jasonwzm looks like that the common files need to be updated.

**Comment by jasonwzm** [comment]
Created https://github.com/istio/common-files/pull/257. Once that merges, please wait for test infra common files to be updated and you can do a retest on this PR.

**Comment by knight42** [comment]
@jasonwzm Thank you!

**Comment by knight42** [comment]
Screen Shot 2020-06-02 at 01 59 29

@jasonwzm an outdated build-tools image is still used in test, could we update it?

**Comment by jasonwzm** [comment]
See https://github.com/istio/test-infra/pull/2685

**Comment by knight42** [comment]
@jasonwzm @ericvn all tests passed now

**Pull request merged**